presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedSmokeTestBase.java

```diff
@@ -1233,4 +1242,108 @@ public void testUnregisterTable()
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (1, 10)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (2, 13)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (3, 5)", 1);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (1, 10), (2, 13), (3, 5)");
+        assertUpdate("DELETE FROM " + tableNameMor + " WHERE id = 1", 1);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (2, 13), (3, 5)");
+
+        dropTable(session, tableNameMor);
+
+        // Test with delete_mode set to copy-on-write
+        String tableNameCor = "test_delete_cor";
+        @Language("RegExp") String errorMessage = "This connector only supports delete where one or more partitions are deleted entirely";
+
+        assertUpdate("CREATE TABLE " + tableNameCor + " (id integer, value integer) WITH (format_version = '2', delete_mode = 'copy-on-write')");
+        assertUpdate("INSERT INTO " + tableNameCor + " VALUES (1, 5)", 1);
+        assertQuery("SELECT * FROM " + tableNameCor, "VALUES (1, 5)");
+        assertQueryFails("DELETE FROM " + tableNameCor + " WHERE value = 5", errorMessage);
+
+        dropTable(session, tableNameCor);
+    }
+
+    @Test
+    public void testDeletePartitionedTable()
+    {
+        Session session = getSession();
+
+        // Test with default delete_mode i.e. merge-on-read:
+
+        String tableNameMor = "test_delete_partitioned_mor";
+
+        assertUpdate("CREATE TABLE " + tableNameMor + " (id integer, value integer) WITH (format_version = '2', partitioning = Array['id'])");
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (1, 10)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (2, 13)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (3, 5)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (4, 13)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (5, 1)", 1);
+        assertUpdate("INSERT INTO " + tableNameMor + " VALUES (6, 1)", 1);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (1, 10), (2, 13), (3, 5), (4, 13), (5, 1), (6, 1)");
+
+        // 1. delete by partitioning column
+        // 2. delete by non-partitioning column
+        assertUpdate("DELETE FROM " + tableNameMor + " WHERE id = 1", 1);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (2, 13), (3, 5), (4, 13), (5, 1), (6, 1)");
+        assertUpdate("DELETE FROM " + tableNameMor + " WHERE value = 13", 2);
+        assertQuery("SELECT * FROM " + tableNameMor, "VALUES (3, 5), (5, 1), (6, 1)");
+
+        dropTable(session, tableNameMor);
+
+        // Test with delete_mode set to copy-on-write
+        String tableNameCor = "test_delete_partitioned_cor";
+        @Language("RegExp") String errorMessage = "This connector only supports delete where one or more partitions are deleted entirely";
```

Comment:
Should the error message be changed to reflect that actually this is now configurable by a table property?